### PR TITLE
SCUMM: Tweak script dumping file management

### PR DESCRIPTION
**Pull request by henke37**
A separate folder per game makes it impossible to mix up scripts from different games.
And add a warning when the file couldn't be opened so that the user is told of the failure.

### CODE REVIEW

File: engines/scumm/resource.cpp [hunk 1]
```diff
@@ -1393,11 +1393,13 @@ void ScummEngine::dumpResource(const char *tag, int id, const byte *ptr, int len
 	else
 		size = READ_BE_UINT32(ptr + 4);
 
-	sprintf(buf, "dumps/%s%d.dmp", tag, id);
+	sprintf(buf, "dumps/%s/%s%d.dmp", _game.gameid, tag, id);
 
-	out.open(buf);
-	if (out.isOpen() == false)
+	out.open(buf, true);
+	if(out.isOpen() == false) {
```
Comment (sev-) [inline engines/scumm/resource.cpp]: Bad formatting.

**Comment by sev-** [comment]
Well, this is suboptimal. As you would always need to create two directories. Why not move htm manually after each scummvm run?

**Comment by henke37** [comment]
The "having to create the folder yourself" part was already addressed with `34a40e6`.

**Comment by sev-** [comment]
Good. Then, we need to change README, the built-in help, and fix the formatting. Then we're good to go

**Comment by henke37** [comment]
You know what? I might as well clean up all usages of `DumpFile` in one sweep.

**Comment by sev-** [comment]
Yes, cleaning up usages in more engines would bring this to consistency.

Also, the README.md file is still unchanged.

**Comment by bluegr** [comment]
Any news on this?

**Comment by henke37** [comment]
The entire `DumpFile` class belongs in the dumpster. The only value it adds is creating folders for you, which should be added as a feature to the `File` class. There is nothing to salvage here.

**Comment by sev-** [comment]
You simply did not pay enough attention to the reason why our `File` class is read-only and `DumpFile` is the only write-capable file class.

**Pull request closed without merging**